include/parse.hpp

```diff
@@ -62,17 +62,64 @@ consteval auto get_current_source_for_parsing() {
         constexpr auto pos = src_sv.find(sep, src_start);
         return pos != std::string_view::npos ? pos : src_sv.size();
     }();
+    
+    static_assert(src_end != src_start, "get_current_source_for_parsing(): error");
     return std::pair{src_start, src_end};
 }
-*/
 
-// Реализуйте семейство функция parse_value
 
-// Шаблонная функция, выполняющая преобразования исходных данных в конкретный тип на основе I-го плейсхолдера
+template <typename T, typename... Types>
+concept IsTypeOneOf = (std::same_as<T, Types> || ...);
+
+// Реализуйте семейство функций parse_value
+template<FixedString str, char format_spec, typename ParsingT>
+requires IsTypeOneOf<ParsingT, int8_t, int16_t, int32_t, int64_t, const int8_t, const int16_t, const int32_t, const int64_t>
+    && (format_spec == 'd' || format_spec == char{})
+consteval ParsingT parse_value() {
+    ParsingT value{};
+    /*
+    Здесь могла быть compile-time обработка ошибки парсинга, например:
+        constexpr bool success = std::from_chars(str.data, str.data + str.size(), value).ec == std::errc{};
+        static_assert(success, "int parsing failed");
+    но любая попытка получить и как-то использовать результат std::from_chars(..) 
+    сразу считается неконстантным выражением, хотя по докам не должно бы.
+
+    есть идеи?
+    */
+    std::from_chars(str.data, str.data + str.size(), value);
+    return value;
```

Comment:
Да, from_chars действительно constexpr, но вот в стандарте constexpr expr требует, чтобы все переменные были constexpr, а в этом коде изменяется value.

Решение совсем не очевидно неочевидное - лямбды, которые сразу вызываются!

```suggestion
    constexpr auto res = []() {
        ParsingT value{};
        auto result = std::from_chars(str.data, str.data + str.size(), value);
        return std::pair{result.ec == std::errc{}, value};
    }();
    
    static_assert(res.first, "could not parse input as a number");
    return res.second;
```